In [1]:
import pandas as pd


# Đọc dữ liệu


In [2]:
df = pd.read_csv('data/world_university_ranking_2023.csv')
df


rank                                               name   
0        1                               University of Oxford  \
1        2                                 Harvard University   
2        3                            University of Cambridge   
3        3                                Stanford University   
4        5              Massachusetts Institute of Technology   
...    ...                                                ...   
1794  1501                               Yamaguchi University   
1795  1501                                 Yanshan University   
1796  1501                                Yeditepe University   
1797  1501  Yuri Gagarin State Technical University of Sar...   
1798  1501                 Zonguldak Bülent Ecevit University   

                 country  number_of_students  student_per_staff   
0         United Kingdom               20965               10.6  \
1          United States               21887                9.6   
2         United Kingdom               20185               11.3   
3          United States               16164                7.1   
4          United States               11415                8.2   
...                  ...                 ...                ...   
1794               Japan               10724               11.1   
1795               China               38649               13.2   
1796              Turkey               17038               28.2   
1797  Russian Federation               28280               34.1   
1798              Turkey               28319               25.4   

      international_students  famale_male_ratio  overall_score   
0                       42.0               48.0           96.4  \
1                       25.0               50.0           95.2   
2                       39.0               47.0           94.8   
3                       24.0               46.0           94.8   
4                       33.0               40.0           94.2   
...                      ...                ...            ...   
1794                     7.0               38.0           18.3   
1795                     1.0               39.0           18.3   
1796                     5.0               55.0           18.3   
1797                     1.0               42.0           18.3   
1798                     8.0               47.0           18.3   

      teaching_score  research_score  citations_score  industry_income_score   
0               92.3            99.7             99.0                   74.9  \
1               94.8            99.0             99.3                   49.5   
2               90.9            99.5             97.0                   54.2   
3               94.2            96.7             99.8                   65.0   
4               90.7            93.6             99.8                   90.9   
...              ...             ...              ...                    ...   
1794            21.3            12.3             13.9                   43.8   
1795            17.2            10.9             18.3                   46.9   
1796            18.8            12.2             10.5                   65.7   
1797            12.5             8.9              7.2                   37.5   
1798            13.2             8.2             23.5                   38.9   

      international_outlook_score  
0                            96.2  
1                            80.5  
2                            95.8  
3                            79.8  
4                            89.3  
...                           ...  
1794                         30.4  
1795                         17.0  
1796                         29.6  
1797                         16.7  
1798                         23.1  

[1799 rows x 13 columns]

In [3]:
df.columns


Index(['rank', 'name', 'country', 'number_of_students', 'student_per_staff',
       'international_students', 'famale_male_ratio', 'overall_score',
       'teaching_score', 'research_score', 'citations_score',
       'industry_income_score', 'international_outlook_score'],
      dtype='object')

In [4]:
df.shape


(1799, 13)

In [5]:
df.dtypes


rank                             int64
name                            object
country                         object
number_of_students               int64
student_per_staff              float64
international_students         float64
famale_male_ratio              float64
overall_score                  float64
teaching_score                 float64
research_score                 float64
citations_score                float64
industry_income_score          float64
international_outlook_score    float64
dtype: object

- Tập dữ liệu được lấy từ `timeshighereducation.com` với các thông tin về xếp hạng các trường đại học trên toàn thế giới năm 2023
- Tập dữ liệu gồm 1799 dòng và 13 cột gồm:
  - rank: xếp hạng của trường đại học
  - name: tên của trường đại học
  - country: quốc gia của trường đại học
  - number_of_students: số lượng sinh viên
  - student_per_staff: số lượng sinh viên trên mỗi giáo viên
  - international_students: phần trăm sinh viên quốc tế
  - female_male_ratio: tỉ lệ sinh viên nữ / nam
  - overall_score: điểm tổng thể
  - teaching_score: điểm giảng dạy
  - research_score: điểm nghiên cứu
  - citations_score: điểm trích dẫn
  - industry_income_score: điểm thu nhập từ ngành công nghiệp
  - international_outlook_score: điểm quan điểm quốc tế

Các điểm đánh giá được giải thích tại [đây](https://www.timeshighereducation.com/student/advice/world-university-rankings-explained).


# Xử lí dữ liệu


In [6]:
df.isnull().sum()


rank                            0
name                            0
country                         0
number_of_students              0
student_per_staff               0
international_students          1
famale_male_ratio              85
overall_score                   0
teaching_score                  0
research_score                  0
citations_score                 0
industry_income_score           0
international_outlook_score     0
dtype: int64

In [7]:
print(len(df['country'].unique()))
# for country in df['country'].unique():
#     print(country)


104


In [8]:
df['country'].value_counts()


- Bảng xếp hạng 1799 trường đại học đến từ 104 quốc gia khác nhau. Trong đó có 5 quốc gia có nhiều trường đại học lọt vào top thế giới nhất, đó là: Hoa Kỳ, Nhật Bản, Anh, Trung Quốc và Nga.
- Nếu như bạn tò mò liệu có trường đại học nào của Việt Nam có thuộc top thế giới hay không? Ta sử dụng câu lệnh:

```python
df[df['country'] == 'Vietnam']
```


In [9]:
df[df['country'] == 'Vietnam']


rank                                            name  country   
422    401                              Duy Tan University  Vietnam  \
489    401                        Ton Duc Thang University  Vietnam   
1190  1001              Vietnam National University, Hanoi  Vietnam   
1608  1501      Hanoi University of Science and Technology  Vietnam   
1612  1501                                  Hue University  Vietnam   
1785  1501  Vietnam National University (Ho Chi Minh City)  Vietnam   

      number_of_students  student_per_staff  international_students   
422                21670               21.9                     0.0  \
489                27878               25.3                     1.0   
1190               48070               15.3                     1.0   
1608               32758               31.1                     0.0   
1612               47590               22.0                     1.0   
1785               75704               21.7                     1.0   

      famale_male_ratio  overall_score  teaching_score  research_score   
422                62.0           44.9            14.4            12.6  \
489                56.0           44.9            13.5            16.0   
1190               63.0           29.7            18.7             9.6   
1608               22.0           18.3            14.9            12.3   
1612               57.0           18.3            13.7             8.1   
1785               46.0           18.3            15.1            10.1   

      citations_score  industry_income_score  international_outlook_score  
422             100.0                   37.7                         48.6  
489              99.2                   39.8                         58.4  
1190             45.9                   38.9                         45.5  
1608             20.2                   43.5                         39.4  
1612             14.5                   37.4                         47.2  
1785             20.7                   41.5                         35.6

In [10]:
# Trong các trường thuộc top, trường có số sinh viên lớn nhất là

df['number_of_students'].max()


460632

In [19]:
county_count = df['country'].value_counts().to_frame("count")

county_count = county_count.rename_axis('country').reset_index()

county_count = county_count.reset_index(drop=True)

county_count


country  count
0         United States    177
1                 Japan    117
2        United Kingdom    103
3                 China     95
4    Russian Federation     76
..                  ...    ...
99           Costa Rica      1
100              Kuwait      1
101          Mozambique      1
102                Oman      1
103          Montenegro      1

[104 rows x 2 columns]

In [21]:
# export to excel without column index

county_count.to_excel('data/country_count_2023.xlsx', index=False)
